main.js

```diff
@@ -2429,7 +2390,80 @@ function startInstance(id, wakeUp) {
                         }
                     }
                     storePids(); // Store all pids to make possible kill them all
-                });
+                };
+                if (config.system.compact && instance.common.compact) {
+                    const vm = new NodeVM({
+                        console: 'inherit',
+                        sandbox: {},
+                        require: {
+                            external: true,
+                            builtin: ['*']
+                        },
+                        nesting: true
+                    });
+
+                    if (!adapterModules[name] && fileNameFull) {
+                        try {
+                            //adapterModules[name] = require(fileNameFull);
+                            adapterModules[name] = vm;
+                        } catch (e) {
+                            logger.error(`host.${hostname} error with ${fileNameFull}: ${JSON.stringify(e)}`);
+                        }
+                    }
+                    if (!adapterModules[name]) {
+                        procs[id].process = cp.fork(fileNameFull, args, {stdio: ['ignore', 'ignore', 'pipe', 'ipc']});
+                    } else {
+                        const _instance = (instance && instance._id && instance.common) ? instance._id.split('.').pop() || 0 : 0;
+                        const logLevel = (instance && instance._id && instance.common) ? instance.common.loglevel || 'info' : 'info';
+                        try {
+                            //procs[id].process = adapterModules[name]({logLevel, compactInstance: _instance, compact: true});
+                            const starterScript =
+                                'module.exports = function (callback) {' +
```

Comment:
I think using backticks (multiline string literals) here would be better for readability.

Reply:
Will think about it ...

Reply:
... would leave it for now that way